Create messages view

Tai-f opened the pull request:
#What
1.generate controller “messages”
2.generate model “model”
3.create migration file
4.configure routes
5.activate “font-awesome”
6.create messages-view by Haml and Sass

Commit:
generate controller "messages"

rails g controller messages

Commit:
generate model "messages"

rails g model “messages”

Commit:
configure routes

Commit:
modify to be activated of scss

Commit:
create migration file

add column ; body, image, created_at,updated_at

Commit:
activate font-awesome

Commit:
create-messages-view

edit source code by Haml and Sass

Tai-f commented:
<img width="1280" alt="create-messages-view" src="https://cloud.githubusercontent.com/assets/22855439/20455892/3f4b2bea-aeaa-11e6-938d-cfbeefbb2a37.png">

madana-i commented on app/assets/stylesheets/application.scss:
```diff
@@ -1,2 +1,6 @@
 @import "./reset";
-= require font-awesome
+@import "messages";
+@import"font-awesome";
```
こういった外部ライブラリは上で読み込むようにしましょう。
ライブラリ内のクラス名は自身で把握していないので、偶然重複してしまった際に、ライブラリ側が優先されてしまって予期せぬ表示くずれが起きる可能性があるためです。

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,224 @@
+    background-color: #38AEF0;
+    text-align: center;
+    min-width: 980px;
+  }
```
messageに関係のないスタイルは別ファイルに記述しましょう。

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,224 @@
+              &before{
+                content: "\f044";
+              }
+              color: white;
```
白色だけでも#FFFFFF #ffffff #fff whiteの4パターン書いてあるので統一しましょう

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,224 @@
+            }
+            i{
+              display: inline-block;
+              font:normal normal normal 14px/1 FontAwesome;
```
:の後ろにスペース入れましょう

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,224 @@
+            color: #fff;
+            font-size: 20px;
+            i{
+              &before{
```
&:beforeですね

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,224 @@
+          content:"";
+          display: block;
+          clear: both;
+        }
```
これはclearfixのようなクラスを作って汎用的に使用できるようにするべきですね

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,224 @@
+      float:left;
+      position: relative;
+      width: calc(100% - 300px);
+      height: 100%;
```
細かいところですが、widthとheightの書き順がバラバラなので統一しましょう

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,224 @@
+          list-style: none;
+          li.chat-message{
+            margin-bottom: 40px;
+            .chat-message__header{
```
この場合は&__headerと書きましょう

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,224 @@
+            margin: 28px 0;
+            padding: 0 20px;
+            line-height: 40px;
+            color: #38aef0;
```
複数回出てくる色は変数化するべきですね。
また、カラーコードの大文字小文字も統一しましょう。

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,224 @@
+              width: 100%;
+              height: 20px;
+              line-height: 16px;
+              padding:15px 10px;
```
ここも半角スペース入れましょう。他にもいくつかあるので修正お願い致します。

madana-i commented on config/routes.rb:
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :messages
```
現在使用中のアクションのみ定義する習慣つけましょう！

Commit:
change the position of "font-awesome"

change the position of "font-awesome"

Commit:
modify messages.scss file

define the name of color which repeats to use.
use “mixin” to clear fix.
modify how to write.(add “space”)
unify the color-code

Commit:
change the position of "font-awesome"

Commit:
change the class name

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,212 @@
+    clear: both;
+  }
+}
+html, body {
```
変数やmixinは汎用的に利用するので、それぞれ別のファイルで管理しましょう！
こちら参考にしてみて下さい！
http://www.mdesign-works.com/blog/web/scss-partials-variables/

Tai-f replied:
$b-color: #38aef0 !default;
$m-color: #434A54 !default;

これを_variables.scss内に記述し
、message.scss内で　
background-color: $b-color;
と記述するとエラーになるのですが、原因はどこにあるのでしょうか。

Tai-f replied:
ご指導通り、エラー解決いたしました。

madana-i commented on config/routes.rb:
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get '/messages' => 'messages#index'
```
resourcesで書き、なおかつ使用中のroutingのみを設定するようお願いします！

madana-i commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,55 @@
+          %textarea#message_body
+          %label.chat-footer__file
+            %i.fa.fa-image
+        %input{ type: "submit", value:"Send"}
```
`:`のあとのスペースと、式展開を使用しない場合はシングルクォーテーションを使用しましょう

Commit:
modify config/routes.rb

to use resources

Commit:
generate mixin.scss and variables.scss

to use “mixin” and “variables”

Commit:
modify index.html.haml

change “” => ‘’  in the last sentence of %input

Commit:
edit messages.scss
